runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/graph/ExecutableStage.java

```diff
@@ -126,7 +126,8 @@ default PTransform toPTransform(String uniqueName) {
     for (SideInputReference sideInput : getSideInputs()) {
       // Side inputs of the ExecutableStage itself can be uniquely identified by inner PTransform
       // name and local name.
-      String outerLocalName = String.format("%s:%s", sideInput.transform(), sideInput.localName());
+      String outerLocalName =
```

Comment:
Harmless but inefficient and confusing - this was using the entire transform's toString instead of the id.